src/sequences_stats.rs

```diff
@@ -0,0 +1,122 @@
+//! Compute statics on sequences from stdin:
+//!   - min: length of shortest sequence
+//!   - max: length of longest sequence
+//!   - average: average length of sequence
+//!   - median: median length of sequence
+//!   - nb_reads: number of reads
+//!   - nb_bases: number of bases
+//!   - n50: N50 of sequences
+//!
+//! ## Usage:
+//!
+//! ```
+//! $ rbt sequences-stats < A.fasta
+//! $ rbt sequences-stats -q < A.fastq
+//! ```
+
+use bio::io::{fasta, fastq};
+
+use quick_error::quick_error;
+
+use std::error::Error;
+use std::io;
+
+quick_error! {
+    #[derive(Debug)]
+    pub enum InputError {
+        NoSequence {
+            description("stdin didn't contains any sequence")
+        }
+
+    }
+}
+
+pub fn stats(fastq: bool) -> Result<(), Box<dyn Error>> {
+    let mut lengths = if fastq {
+        fastq_lengths()
+    } else {
+        fasta_lengths()
+    };
+
+    if lengths.is_empty() {
+        return Err(Box::new(InputError::NoSequence));
+    }
+    // Sort lengths one time
+    lengths.sort();
+
+    let nb_bases = lengths.iter().sum::<usize>();
+
+    println!(
+        "min: {min}
+max: {max}
+average: {average}
+mediane: {mediane}
+number of reads: {nb_reads}
+number of bases: {nb_bases}
+n50: {n50}",
```

Comment:
Please mention that output is in YAML format in the help text.

Reply:
It's done